src/Helpers/IPFSClient.ts

```diff
@@ -0,0 +1,98 @@
+import * as FormData from 'form-data'
+import fetch from 'node-fetch'
+import * as str from 'string-to-stream'
+
+interface IPFSConfiguration {
+  readonly ipfsUrl: string
+}
+
+interface FetchConfiguration {
+  readonly timeout: number
+}
+
+enum Type {
+  File = 'File',
+  Directory = 'Directory',
+}
+
+interface IPFSObject {
+  readonly Name?: string
+  readonly Hash: string
+  readonly Size?: number
+  readonly Type?: Type
+  readonly Links?: ReadonlyArray<IPFSObject>
+}
+
+interface LSResult {
+  Arguments: {
+    [key: string]: string
+  }
+  Objects: {
+    [key: string]: IPFSObject
+  }
+}
+
+type addFileToDirectory = (directoryHash: string, fileName: string, filehash: string) => Promise<IPFSObject>
+
+type addText = (text: string, x?: FetchConfiguration) => Promise<IPFSObject>
+
+type cat = (s: string, x?: FetchConfiguration) => Promise<string>
+
+type createEmptyDirectory = () => Promise<IPFSObject>
+
+type getDirectoryFileHashes = (s: string) => Promise<ReadonlyArray<string>>
+
+type ls = (s: string) => Promise<LSResult>
+
+export class IPFS {
+  private readonly url: string
+
+  constructor(configuration: IPFSConfiguration) {
+    this.url = configuration.ipfsUrl
+  }
+
+  addFileToDirectory: addFileToDirectory = async (directoryHash, fileName, fileHash) => {
+    const response = await fetch(
+      `${this.url}/api/v0/object/patch/add-link?arg=${directoryHash}&arg=${fileName}&arg=${fileHash}`
+    )
+    return await response.json()
+  }
+
+  addText: addText = async (text: string, fetchConfiguration) => {
+    const formData = new FormData() // { maxDataSize: 20971520 }
+
+    formData.append('file', str(text), {
+      knownLength: text.length,
+      filename: 'file',
+      contentType: 'plain/text',
+    })
+
+    const response = await fetch(`${this.url}/api/v0/add`, {
+      ...fetchConfiguration,
+      method: 'post',
+      body: formData,
+    })
+
+    return await response.json()
+  }
+
+  cat: cat = async (hash, fetchOptions) => {
+    const response = await fetch(`${this.url}/api/v0/cat?arg=${hash}`, fetchOptions)
+    return response.text()
```

Comment:
Why does cat return `response.text()` instead of `response.json()`?

Reply:
Cool, so I think we should be able to change it now to `response.json()` without breaking anything I think.

Reply:
Changed `cat` type to return Response["buffer"] from node-fetch as you suggested, ts then marked the `cat` function red, I then changed the cat function to `response.json()` and it removed the red, so it seems to like the change.